### PR TITLE
Minor updates

### PR DESCRIPTION
**Pull request by sourabhdalvi**
* Refactoring hybrid range constraint code
* Updating methods to save sparse axis arrays in the right order
* Updating the indexing od piece linear cost variables 
* Adding hybrid test simulation

### CODE REVIEW

File: src/devices_models/devices/common/cost_functions.jl [hunk 1]
```diff
@@ -270,15 +270,14 @@ function pwl_gencost_sos!(
     for i in 1:length(cost_data)
         pwlvars[i] = JuMP.@variable(
             container.JuMPmodel,
-            # TODO: Redefine the base_name for PWL vars.
-            # base_name = "{$(variable)}_{sos}",
+            base_name = "{$(variable)}_{sos_$(i)}",
```
Comment (jd-lara) [inline src/devices_models/devices/common/cost_functions.jl:273]: Did you test how does this look?

**Review by jd-lara — COMMENTED** [review]
Check the printing of the sos please

**Review by jd-lara — CHANGES_REQUESTED** [review]
@sourabhdalvi can you make a test with a two-stage model Hybrid please?

File: src/devices_models/devices/common/cost_functions.jl (hunk 1 above)
Comment (sourabhdalvi) [reply src/devices_models/devices/common/cost_functions.jl:273]: this is how the variable name looks like , `ActivePowerVariable_ThermalStandard_{Alta, 2}_{pwl_3}` `ActivePowerVariable_ThermalStandard_{Alta, 2}_{sos_2}`

File: src/utils/jump_utils.jl [hunk 2]
```diff
@@ -66,16 +66,27 @@ function to_array(array::DenseAxisArray{<:Number})
 end
 
 function to_array(array::SparseAxisArray)
```
Comment (jd-lara) [inline src/utils/jump_utils.jl:68]: @daniel-thom there are some coming changes to this function to support Hybrid Systems

File: src/utils/jump_utils.jl [hunk 3]
```diff
@@ -66,16 +66,27 @@ function to_array(array::DenseAxisArray{<:Number})
...
+    final_column_names = Vector{Symbol}()
     for (ix, col) in enumerate(columns)
-        res = values(filter(v -> first(v)[[1, 3]] == col, array.data))
-        tmp_data[col] = jump_value.(res)
+        filtered_vars =
+            collect(filter(v -> (first(v)[[1, 2]] == col) && (last(v) != 0), array.data))
+        sorted_vars = sort(filtered_vars, by = x -> x[1][3])
+        vars = last.(sorted_vars)
+        if !isempty(vars)
+            tmp_data[Symbol(col...)] = PSI._jump_value.(vars)
```
Comment (jd-lara) [inline src/utils/jump_utils.jl:79]: @sourabhdalvi you don't need the `PSI.` here. I overlooked it.

File: src/utils/jump_utils.jl (hunk 2 above)
Comment (jd-lara) [inline src/utils/jump_utils.jl:68]:
```suggestion
function to_array(array::SparseAxisArray)
    columns = Set()
    timesteps = Set{Int}()
    for k in keys(array.data)
        push!(columns, (k[1], k[2]))
        push!(timesteps, k[3])
    end

    data = Array{Float64, 2}(
        undef,
        length(timesteps),
        length(columns),
    )

    for (ix, col) in enumerate(columns), t in timesteps
        data[t, ix] = array.data[(col..., t)]
    end

    return data
end
```

**Pull request merged**